### PR TITLE
add basic unit tests for render command

### PR DESCRIPTION
**Pull request by mfojtik**
This will exercise the happy path and no-args handling. We can extend this with actually checking the output assets to have data we want. For now this proofs the rendering is working.

/cc @sttts

### CODE REVIEW

File: cmd/cluster-kube-controller-manager-operator/render/options/generic.go [hunk 1]
```diff
@@ -44,10 +44,10 @@ func (o *GenericOptions) Complete() error {
 // Validate verifies the inputs.
 func (o *GenericOptions) Validate() error {
 	if len(o.AssetInputDir) == 0 {
-		return errors.New("missing required flag: --asset-output-dir")
+		return errors.New("missing required flag: --asset-input-dir")
```
Comment (mfojtik) [inline cmd/cluster-kube-controller-manager-operator/render/options/generic.go]: @sttts typo found during testing ;-)

File: cmd/cluster-kube-controller-manager-operator/render/options/generic.go [hunk 2]
```diff
@@ -44,10 +44,10 @@ func (o *GenericOptions) Complete() error {
...
 	}
 	if len(o.AssetOutputDir) == 0 {
-		return errors.New("missing required flag: --asset-input-dir")
+		return errors.New("missing required flag: --asset-output-dir")
```
Comment (sttts) [inline cmd/cluster-kube-controller-manager-operator/render/options/generic.go]: :+1:

File: cmd/cluster-kube-controller-manager-operator/render/render.go [hunk 3]
```diff
@@ -44,6 +51,16 @@ func NewRenderCommand() *cobra.Command {
 	return cmd
 }
 
+func (r *renderOpts) handleError(cmd *cobra.Command, err error) {
```
Comment (mfojtik) [inline cmd/cluster-kube-controller-manager-operator/render/render.go]: @sttts candidate to move to generic stuff

File: cmd/cluster-kube-controller-manager-operator/render/render.go [hunk 4]
```diff
@@ -20,20 +21,26 @@ const (
...
+	renderOpts := &renderOpts{errOut: errOut}
 	cmd := &cobra.Command{
 		Use:   "render",
 		Short: "Render kubernetes controller manager bootstrap manifests, secrets and configMaps",
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := renderOpts.Validate(); err != nil {
-				glog.Fatal(err)
+			var err error
+			defer func() {
+				renderOpts.handleError(cmd, err)
```
Comment (sttts) [inline cmd/cluster-kube-controller-manager-operator/render/render.go]: clever

File: cmd/cluster-kube-controller-manager-operator/render/render.go [hunk 5]
```diff
@@ -20,20 +21,26 @@ const (
...
+			}()
+			if err = renderOpts.Validate(); err != nil {
```
Comment (sttts) [inline cmd/cluster-kube-controller-manager-operator/render/render.go]: something like `must(renderOpts.Validate())` would be simpler and clearer.

File: cmd/cluster-kube-controller-manager-operator/render/render_test.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	if len(errBytes) == 0 {
+		return c, nil
+	}
+	return c, errors.New(string(errBytes))
+}
+
+func setupAssetOutputDir(testName string) (teardown func(), outputDir string, err error) {
+	outputDir, err = ioutil.TempDir("", testName)
+	if err != nil {
+		return
```
Comment (sttts) [inline cmd/cluster-kube-controller-manager-operator/render/render_test.go]: `nil, "", err`

**Comment by sttts** [comment]
overall lgtm. some nits.

File: cmd/cluster-kube-controller-manager-operator/render/render.go (hunk 4 above)
Comment (mfojtik) [reply cmd/cluster-kube-controller-manager-operator/render/render.go]: don't see other option to shim into cobra command ;-)

**Comment by mfojtik** [comment]
@sttts fixed

File: cmd/cluster-kube-controller-manager-operator/render/render.go [hunk 7]
```diff
@@ -20,27 +21,33 @@ const (
...
 	cmd := &cobra.Command{
 		Use:   "render",
 		Short: "Render kubernetes controller manager bootstrap manifests, secrets and configMaps",
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := renderOpts.Validate(); err != nil {
-				glog.Fatal(err)
-			}
-			if err := renderOpts.Run(); err != nil {
-				glog.Fatal(err)
+			must := func(fn func() error) {
```
Comment (sttts) [inline cmd/cluster-kube-controller-manager-operator/render/render.go]: why not the error directly? Why the indirection with a func?

**Comment by sttts** [comment]
Last nit. otherwise lgtm.

File: cmd/cluster-kube-controller-manager-operator/render/render.go (hunk 7 above)
Comment (mfojtik) [reply cmd/cluster-kube-controller-manager-operator/render/render.go]: @sttts error where? the Execute() won't propagate that error (Run does not return errors).

Comment (sttts) [reply cmd/cluster-kube-controller-manager-operator/render/render.go]:
```
must := func(error) {
  if err == nil {
    return
  }
  ....
}

must(renderOpts.Validate())
```

**Comment by sttts** [comment]
/lgtm

we can iterate on nits.

**Pull request merged**